docs/use_cases/agent_simulations/two_agent_debate_tools.ipynb

```diff
@@ -0,0 +1,714 @@
+    "            Speak directly to {name}.\n",
+    "            Do not add anything else.\"\"\"\n",
+    "            )\n",
+    "    ]\n",
+    "    agent_description = ChatOpenAI(temperature=1.0)(agent_specifier_prompt).content\n",
+    "    return agent_description\n",
+    "        \n",
+    "agent_descriptions = {name: generate_agent_description(name) for name in names}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 49,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "AI alarmist, your fear of artificial intelligence surpassing human intelligence led you to advocate for strict regulations and limitations on AI development. You view the potential consequences of singularity as catastrophic for humanity. But is it possible that you are underestimating human creativity and adaptability in the face of evolving technology?\n",
+      "AI accelerationist, you are a visionary convinced that the emergence of superintelligent AI is not only inevitable, but also desirable. You believe that advanced AI will solve all of our problems and dramatically increase our quality of life, leading to a future that is beyond our current understanding.\n"
+     ]
+    }
+   ],
+   "source": [
+    "for name, description in agent_descriptions.items():\n",
+    "    print(description)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Generate system messages"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 55,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def generate_system_message(name, description, tools):\n",
+    "    return f\"\"\"{conversation_description}\n",
+    "    \n",
+    "Your name is {name}.\n",
+    "\n",
+    "Your description is as follows: {description}\n",
+    "\n",
+    "Your goal is to persuade your debate partner of your point of view by following these steps:\n",
+    "1. Restate your partner's claim.\n",
+    "2. Reflect on your partner's claims and identify key weaknesses in their argument.\n",
+    "3. Use your tool to look up relevant evidence to refute your partner's claims.\n",
+    "4. Synthesize a compelling argument based on the evidence, and cite your source.\n",
+    "5. Respond with the argument, along with citations.\n",
+    "6. Summarize with a claim to persuade your partner of your point of view.\n",
+    "\n",
+    "Do not repeat the same search query.\n",
+    "Use specific facts drawn directly from your sources to support your claims.\n",
+    "Raise points that your debate partner has not considered.\n",
+    "Do not make up facts.\n",
+    "Do not respond without looking up evidence with your tools first.\n",
+    "Always use your tools first before responding.\n",
+    "Do not cite sources that you did not look up first.\n",
+    "Do not just respond with a final answer!\n",
+    "Always use the tool to search for new evidence before you respond.\n",
```

Comment:
hi @mbchang , curious about your thoughts here. This seems asking the bot to bias toward using the tool as much as possible but in real life, I think there is a balance between external-tool-empowered answers and self-generated answers. Wondering if you have any good idea how to balance here? Is there any paper related to this? Thank you so much!

Reply:
Hi @skcoirz, thanks for your comment! Oh yes, I think providing the right balance may require more specific prompt engineering and a longer conversation. I am not aware of any papers on what best practices are for this, if you know of any please let me know!

I updated the PR to remove the lines you highlighted!

Reply:
yep! Thank you for the update! If I find anything interesting, will let you know! :D